From foo@baz.org Sun Jun  5 14:33:21 2005
Date: Mon, 20 Jun 2005 21:15:16 -0700
To: Greg KH <gregkh@suse.de>
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: devfs: Remove devfs_mk_symlink() function from the kernel tree

Removes the devfs_mk_symlink() function and all callers of it.

Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

 arch/um/drivers/ubd_kern.c      |    7 -------
 include/linux/devfs_fs_kernel.h |    4 ----
 2 files changed, 11 deletions(-)

---
--- gregkh-2.6.orig/arch/um/drivers/ubd_kern.c
+++ gregkh-2.6/arch/um/drivers/ubd_kern.c
@@ -620,7 +620,6 @@ static int ubd_new_disk(int major, u64 s
 			
 {
 	struct gendisk *disk;
-	char from[sizeof("ubd/nnnnn\0")], to[sizeof("discnnnnn/disc\0")];
 	int err;
 
 	disk = alloc_disk(1 << UBD_SHIFT);
@@ -634,12 +633,6 @@ static int ubd_new_disk(int major, u64 s
 	if(major == MAJOR_NR){
 		sprintf(disk->disk_name, "ubd%c", 'a' + unit);
 		sprintf(disk->devfs_name, "ubd/disc%d", unit);
-		sprintf(from, "ubd/%d", unit);
-		sprintf(to, "disc%d/disc", unit);
-		err = devfs_mk_symlink(from, to);
-		if(err)
-			printk("ubd_new_disk failed to make link from %s to "
-			       "%s, error = %d\n", from, to, err);
 	}
 	else {
 		sprintf(disk->disk_name, "ubd_fake%d", unit);
--- gregkh-2.6.orig/include/linux/devfs_fs_kernel.h
+++ gregkh-2.6/include/linux/devfs_fs_kernel.h
@@ -15,10 +15,6 @@ static inline int devfs_mk_cdev(dev_t de
 {
 	return 0;
 }
-static inline int devfs_mk_symlink(const char *name, const char *link)
-{
-	return 0;
-}
 static inline void devfs_remove(const char *fmt, ...)
 {
 }
